system/Model.php

```diff
@@ -592,9 +592,9 @@ protected function doErrors()
      */
     public function getIdValue($row)
     {
-        if (is_object($row) && isset($row->{$this->primaryKey})) {
+        if (is_object($row)) {
             // Get the raw primary key value of the Entity.
-            if ($row instanceof Entity) {
+            if ($row instanceof Entity && $row->{$this->primaryKey} !== null) {
```

Comment:
In the Entity, if the attribute key is mapped, it will return `false` for `isset()`, since only the mapped key name will be recognized.

So if we have:
```php
$attributes = [
    'id' => null,
];
$datamap = [
    'mapped_id' => 'id',
];

isset($entity->id); // will return false
isset($entity->mapped_id); // will return true
```

The `$row->{$this->primaryKey}` will use `__get()` which handle data mapping. This will return `null` when either the attribute key or mapped attribute key are not found.

Reply:
Thanks for the explanation. Can we change the comment above to reflect this? Just a short one will do.